src/wp-includes/functions.php

```diff
@@ -5661,6 +5661,10 @@ function iis7_supports_permalinks() {
  * @return int 0 means nothing is wrong, greater than 0 means something was wrong.
  */
 function validate_file( $file, $allowed_files = array() ) {
+	if ( ! is_scalar( $file ) || '' === $file ) {
```

Comment:
Why `is_scalar()` instead of `is_string()`?  
It is possible that some code somewhere is passing in an integer that could be in the `$allowed_files` which then could pass the checks below to return `3`. Using `is_scalar()` is strict but also avoids a b/c.